ddtrace/tracer/tracer_test.go

```diff
@@ -351,6 +351,11 @@ func TestSamplingDecision(t *testing.T) {
 
 	t.Run("client_dropped", func(t *testing.T) {
 		tracer, _, _, stop := startTestTracer(t)
+		defer func() {
```

Comment:
We should do these same checks for the single spans code too. Since no one is using it, it's fine if we do that in a follow-up PR though.